server/app/controller/question.js

```diff
@@ -1,53 +1,84 @@
-
-
-const Questions = [];
-// Get all Questions
-const getAll = () => Promise.resolve(Questions);
-
-// Get one question
+import db from '../db/dbconnect';
+/** Get all question
+ * @return {obj}
+ * @public
+*/
+const getAll = () => {
+  return db.any('select * from questions')
+    .then((data) => {
+      return Promise.resolve(data);
+    });
+};
+/**  Get one question
+ * @param {string}
+ * @return {obj}
+ * @public
+*/
 const getOne = (id) => {
-  const questionId = parseInt(id, 10);
-  const question = Questions.find(questn => questn.id === questionId);
-  if (!question) return Promise.reject(new Error('pass valid id'));
-  return Promise.resolve(question);
+  const questionId = parseFloat(id);
+  return db.task(t => t.batch([
+    t.one('SELECT id, question FROM questions WHERE id =$1', questionId),
+    t.any('SELECT answer_id,answer FROM answers WHERE answers.question_id =$1', questionId),
+  ])).spread((question, answers) => {
+    question.answers = answers;
+    return question;
+  })
+    .then((data) => {
+      return Promise.resolve(data);
+    });
 };
-
-//  POST a question
+/**  POST a question
+ * @param {string}
+ * @return {obj}
+ * @public
+*/
 const postQuestion = (question) => {
-  const post = {
-    id: Questions.length + 1,
-    answers: [],
-    question,
-  };
   if (!question) return Promise.reject(new Error('post a question'));
-
-  Questions.push(post);
-  console.log(post);
-  return Promise.resolve(post);
+  return db.one('INSERT INTO questions (question) VALUES($1) RETURNING id', question)
+    .then((data) => {
+      return Promise.resolve(data);
+    });
 };
-
-// POST answer
+/**  POST answer
+ * @param {string}
+ * @return {obj}
+ * @public
+*/
 const postAnswer = (questionId, answer) => {
   const id = parseFloat(questionId);
```

Comment:
Do you intend to have a floating number as a question id? I suggest you replace this with parseInt and use only integer values for your id field.